walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,97 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        Collection<String> mandatoryAttendees = request.getAttendees();
+        Collection<String> allAttendees = request.getAllAttendees();
+        long duration = request.getDuration();
+
+        Collection<Event> mandatoryAttendeeEvents = getEventsForAttendees(mandatoryAttendees, events);
+        ArrayList<TimeRange> mandatoryAttendeesSlots = getPossibleSlots(mandatoryAttendeeEvents, mandatoryAttendees, duration);
```

Comment:
Prefer to declare variables using the interface type (List) rather than the concrete type (ArrayList)